src/main/java/logisticspipes/renderer/LogisticsRenderPipe.java

```diff
@@ -78,6 +78,10 @@ public class LogisticsRenderPipe extends TileEntitySpecialRenderer {
     private final RenderBlocks renderBlocks = new RenderBlocks();
     private final IBCRenderTESR bcRenderer = SimpleServiceLocator.buildCraftProxy.getBCRenderTESR();
 
+    // To reduce allocations on the hot path, we cache a single instance and reset it instead of
+    // allocating hundreds of temp objects every frame.
+    private final LPPosition pos = new LPPosition(0, 0, 0);
```

Comment:
will it work with Angelica's Multi-Draw, which uses some parallel threads to render stuff?

Reply:
I know that Angelica uses ISBRHs across threads if they are marked up. Does it do the same for TESRs? That would probably be a problem for LP anyway since the TESR is not thread-safe.

I could make one `LPPosition` variable in the function. That would be thread-safe. And HotSpot's escape analysis might be able optimize out the allocation in any case.

Reply:
only ISBRH's are assemebled off thread, when opted into as thread safe.

TESRs are always run on the main thread -- only the main thread can make GLxx calls